lib/rubyXL/convenience_methods.rb

```diff
@@ -179,6 +183,25 @@ def modify_border(style_index, direction, weight)
       register_new_xf(new_xf, style_index)
     end
 
+    def modify_number_format(style_index, format_code)
```

Comment:
It needs to be more complex than this. If you look at `get_number_format_by_id`, you will see that there is a hardcoded 'omnipresent' `DEFAULT_NUMBER_FORMATS` hash that participates in the search but not modification. And also the IDs from that hash cannot be reused... So, your idea in general is viable, but the patch in its current form cannot be accepted.